### PR TITLE
Adding client side test coverage.

### LINKED ISSUE #663: Client side test coverage
I'm working on client side test coverage in my own fork.  I'm happy to create PR's, but wasn't sure if the extra coverage would be useful or not to others.. any interest?

edit: this is on the 0.4.0 branch

### PR DESCRIPTION
**Pull request by cdriscol**
Fixes #663.

* Adding __coverage/__ to .gitignore
* Adding client test coverage for:
 * Articles controller
 * Chat controller
 * Menus service
 * Header controller
 * Password controller
 * Authentication controller
* Small changes on Menu service
 * Tabs changed to 2 spaces
 * Removed [unreachable line](https://github.com/meanjs/mean/pull/728/files#diff-1e5dbd4f8e8c78fcdf6081de540f09d1L46)
 * Fixing [typeo](https://github.com/meanjs/mean/pull/728/files#diff-1e5dbd4f8e8c78fcdf6081de540f09d1L40) on error message.

### CODE REVIEW

**Comment by cdriscol** [comment]
All (100) client tests are passing!

**Comment by cdriscol** [comment]
Just realized I made the same changes as #724 in fixing the angular-file-upload paths.. I can resolve that if/when that PR gets merged into 0.4.0.

File: config/assets/default.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ module.exports = {
 				'public/lib/angular-ui-router/release/angular-ui-router.js',
 				'public/lib/angular-ui-utils/ui-utils.js',
 				'public/lib/angular-bootstrap/ui-bootstrap-tpls.js',
-				'public/lib/angular-file-upload/angular-file-upload.js'
+				'public/lib/angular-file-upload/src/module.js'
```
Comment (ilanbiala) [inline config/assets/default.js]: Let's leave this out, we'll just statically use 1.1.5 until we bump a lot of the deps.

**Comment by ilanbiala** [comment]
LGTM other than the one comment above. @cdriscol if you can resolve that, I'll merge this in right after.

File: config/assets/default.js (hunk 1 above)
Comment (ilanbiala) [inline config/assets/default.js]: This has been taken care of as of #729, so once you undo this change and squash, I'll merge this in.

**Comment by codydaig** [comment]
LGTM

**Comment by lirantal** [comment]
Looks good, all yours @ilanbiala

**Pull request merged**